systemtest/loadgen/config/config.go

```diff
@@ -52,29 +53,42 @@ func init() {
 	flag.StringVar(&Config.APIKey, "api-key", "", "API key for APM Server")
 	flag.BoolVar(&Config.Secure, "secure", false, "validate the remote server TLS certificates")
 	flag.Func(
-		"max-rate",
-		"Max event rate as epm or eps with burst size=max(1000, 2*eps), <= 0 values evaluate to Inf (default 0epm)",
+		"event-rate",
+		"Event rate in format of {burst}/{interval}. For example, 1000/1s or 1000/5s, <= 0 values evaluate to Inf (default 0/s)",
 		func(rate string) error {
-			errStr := "invalid value %s for -max-rate, valid examples: 5eps or 10epm"
-			r := strings.Split(rate, "ep")
+			errStr := "invalid value %s for -event-rate, valid examples: 1000/1s or 1000/5s"
+			r := strings.Split(rate, "/")
 			if len(r) != 2 {
 				return fmt.Errorf(errStr, rate)
 			}
-			rateVal, err := strconv.Atoi(r[0])
-			if err != nil {
+			burst, err := strconv.Atoi(r[0])
+			if err != nil || burst < 0 {
 				return fmt.Errorf(errStr, rate)
 			}
-			switch r[1] {
+			n := r[1][:len(r[1])-1]
+			if n == "" {
+				n = "1"
+			}
+			u := r[1][len(r[1])-1:]
+			t, err := strconv.Atoi(n)
+			if err != nil || t <= 0 {
+				return fmt.Errorf(errStr, rate)
+			}
+			// Interval, EPS = 0
+			if burst == 0 {
+				return nil
+			}
+			switch u {
 			case "s":
-				Config.MaxEPM = rateVal * 60
+				Config.Interval = t
 			case "m":
-				Config.MaxEPM = rateVal
+				Config.Interval = t * 60
 			default:
 				return fmt.Errorf(errStr, rate)
 			}
+			Config.EPS = burst / Config.Interval
 			return nil
 		})
```

Comment:
IMO this code has become sufficiently complicated that we should create a custom flag:

```go
type RateFlag struct {
        burst    int
        interval time.Duration
}

func (f *RateFlag) String() string {
        return fmt.Sprintf("%d/%s", f.burst, f.interval)
}

func (f *RateFlag) Set(s string) error {
        before, after, ok := strings.Cut(s, "/")
        if !ok || before == "" || after == "" {
                return fmt.Errorf("invalid rate %q, expected format burst/duration", s)
        }

        burst, err := strconv.Atoi(before)
        if err != nil {
                return fmt.Errorf("invalid burst %s in event rate: %w", before, err)
        }

        if !(after[0] >= '0' && after[0] <= '9') {
                after = "1" + after
        }
        interval, err := time.ParseDuration(after)
        if err != nil {
                return fmt.Errorf("invalid interval %q in event rate: %w", after, err)
        }
        if interval <= 0 {
                return fmt.Errorf("invalid interval %q, must be positive", after)
        }

        f.burst = burst
        f.interval = interval
        return nil
}
```

(May require some changes, I haven't fully tested that.)

Reply:
It works like a charm 👍